### ISSUE TITLE
scopelint reports false positives

### ISSUE THREAD

**Issue by zlozano** [golangci/golangci-lint#281]
I recently pulled down the latest golangci-lint, and it is reporting false positives for scopelint on my table test.  e.g.

`
   func TestSomething(t *testing.T) {
        tc := []struct {
		Name        string
		Expected  string
	}{
		{
			Name:       "foo",
			Expected: "bar",
		},
		{
			Name:       "fizz",
			Expected: "buzz",
		},
	}

	for _, tt := range tc {
		t.Run(tt.Name, func(t *testing.T) {
			result := thingUnderTest(...)
                         if result != tt.Expected {
                             t.Fatal("failed.")
                         }
		})
	}
   }
`

In this case since the testing closure is executed immediately in the same iteration, the mistake discussed here: https://github.com/golang/go/wiki/CommonMistakes is not relevant.

**Comment by powerman** [comment]
Another example of same issue:
```go
for _, val := range vals {
    i := sort.Search(…, func(i int) bool { … val … })
}
```
Probably same happens with `sort.Slice` but I didn't tested this.

**Comment by powerman** [comment]
Yes, it does reproduce.

**Comment by mastertinner** [comment]
I just created an issue for scopelint. I think this one here can be closed because it's not a golangci-lint issue.
